### PR TITLE
Feedback Sharing

### PR DESCRIPTION
**Pull request by feng-eric**
- Added ability to send feedback email to select schools as well as sending feedback email to all schools
- Added file for AWS SES config (Note: Need to add AWS_ACCESS_KEY_ID, AWS_SECRET_ACCESS_KEY, AWS_REGION) to .env
- Added ability to filter performances by schoolIDs
- Added ability to get schools by IDs
- Added returning school data with school contacts
- Changed API routes to plural for consistency

TODO: 
- Format the email body -> to be done after awards are done and a design for what email should look like?
- What should happen if one email fails to send when sending multiple emails?
- Only retrieve performances when all the judging adjudications are submitted.

### CODE REVIEW

**Comment by thepenguinco** [comment]
I'll review this in a couple hours when I wake up.

**Review by thepenguinco — APPROVED** [review]
LGTM, let's just get another review on this before merging.
I'll add the environment variables into staging and prod environments.

File: pages/api/feedback/index.js
Comment (thepenguinco) [inline pages/api/feedback/index.js:48]: Let's get this into an environment variable as well.

**Comment by thepenguinco** [comment]
Last thing, can you update the .env.sample file with the new required environment variables.

Thanks!

File: aws/index.js
Comment (OustanDing) [inline aws/index.js:1]: Use `import`?

File: aws/index.js
Comment (OustanDing) [inline aws/index.js:13]: Won't this create a new object on every import? Is that intentional?

File: pages/api/feedback/index.js
Comment (OustanDing) [inline pages/api/feedback/index.js:3]: Are these relative imports okay? @Anish-Agnihotri

File: pages/api/feedback/index.js
Comment (OustanDing) [inline pages/api/feedback/index.js:17]:
```suggestion
    if (schools.length === 0) return res.status(400).json({ error: 'No schools to email feedback to.' });
```

Comment (OustanDing) [reply pages/api/feedback/index.js:17]: Might have some linting issues on this line.

File: pages/api/feedback/index.js
Comment (OustanDing) [inline pages/api/feedback/index.js:30]:
```suggestion
    const performancesMap = {};
```

Not reassigning, so `const` should be fine.

File: pages/api/feedback/index.js
Comment (OustanDing) [inline pages/api/feedback/index.js:42]:
```suggestion
      if (school.id in performancesMap && 'contacts' in school && school.contacts.length > 0) {
```

File: pages/api/performances/collect.js
Comment (OustanDing) [inline pages/api/performances/collect.js:16]:
```suggestion
    const filter = { event_id: parseInt(eventID) };
```

File: pages/api/schools/collect.js
Comment (OustanDing) [inline pages/api/schools/collect.js:12]:
```suggestion
    const filter = {};
```

**Review by OustanDing — CHANGES_REQUESTED** [review]
A couple of comments.

File: aws/index.js
Comment (feng-eric) [reply aws/index.js:13]: I followed their documentation so I'm not exactly sure https://nodemailer.com/transports/ses/

Comment (feng-eric) [reply aws/index.js:13]: Are you saying it should be without the `new`?

Comment (OustanDing) [reply aws/index.js:13]: Hm, never mind. I don't think it will create a new one. My bad. This is fine.

**Review by OustanDing — APPROVED** [review]

File: pages/api/feedback/index.js
Comment (Anish-Agnihotri) [reply pages/api/feedback/index.js:3]: Think we should migrate to absolute, and if there isn't one for `@page` and that in `jsconfig.json`.

File: pages/api/feedback/index.js
Comment (Anish-Agnihotri) [inline pages/api/feedback/index.js:47]: Instead of processing these in a loop, we may want to push the actual requests into an array, and then `Promise.all()` the array execution to parallelize?

**Review by Anish-Agnihotri — APPROVED** [review]

File: pages/api/feedback/index.js
Comment (feng-eric) [reply pages/api/feedback/index.js]: good idea, I made that change

**Pull request merged**
